packages/core/src/integration-tests/index.test.ts

```diff
@@ -405,6 +405,41 @@ describe("Pilot Integration Tests", () => {
         pilot.perform("Action with write error"),
       ).resolves.not.toThrow();
     });
+
+      it("should add error and not add result for next perform step", async () => {
+          let promptParam: string = "";
+          mockPromptHandler.runPrompt
+              .mockResolvedValueOnce(
+                  'throw new Error("Element not found");',
+              )
+              .mockImplementationOnce((prompt, _snapshot) => {
+                  promptParam = prompt;
+                  return Promise.resolve("// No operation");
+              });
+
+          await pilot.perform("Tap on a non-existent button");
+
+          expect(promptParam).toContain("Element not found");
+          expect(promptParam).toMatchSnapshot();
+      });
+
+      it("should add result and not add error for next perform step", async () => {
+          let promptParam: string = "";
+          mockPromptHandler.runPrompt
+              .mockImplementationOnce((prompt, _snapshot) => {
+                  promptParam = prompt;
+                  return Promise.resolve("Operation successful");  // Return a successful result here
```

Comment:
I think this will throw an error because the code evaluator tries to evaluate this and that's not a code. Try to resolve with a comment (e.g. see next mockImplementation)